### PR TITLE
Keep code aligned with latest repo sources

### PR DESCRIPTION
**Pull request by alexpalms**
This PR substitutes all hand coded snippets in the docs with their real source code in the different repos (arena and agents).

The CICD workflow has been modified so that the reusable build can be called (as a test) from arena and agents repos every time there is a new PR (or a commit on main), in this way we can make sure the referenced source code files that are imported here are still there (otherwise the CICD will throw an error).

### CODE REVIEW

**Comment by alexpalms** [comment]
One additional thing I would like to implement is that, everytime there is a push to main in arena and or agents, a new docs build+deploy is triggered. But I am not sure if this can be done with the resuable workflow approach, i.e. adding `on: workflow call:` for the deploy CICD, wdyt @discordianfish ?

File: .github/workflows/reusable-build.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - uses: actions/checkout@v3
+        with:
+          submodules: true  # Fetch Hugo themes (true OR recursive)
+          fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.112.3'
+          extended: true
```
Comment (alexpalms) [inline .github/workflows/reusable-build.yml:17]: @discordianfish note that I am re-activating this extended option. Locally it works fine with the same hugo engine version. Is it ok for you? Any specific reason why you commented it out?

File: content/envs/_index.en.md [hunk 2]
```diff
@@ -105,35 +105,7 @@ DIAMBRA Arena Environments usage follows the standard RL interaction framework:
...
-
-     # Episode end (Done condition) check
-     if done:
-         observation = env.reset()
-         break
-
- # Environment close
- env.close()
-```
+{{< github_code "https://raw.githubusercontent.com/diambra/arena/main/examples/diambra_arena_gist.py" >}}
```
Comment (alexpalms) [inline content/envs/_index.en.md:108]: @discordianfish this is an example on how all hand coded snippets have been removed from the docs source code, and automatically obtained from the repos.

**Review by discordianfish — APPROVED** [review]
LGTM

File: .github/workflows/reusable-build.yml (hunk 1 above)
Comment (alexpalms) [reply .github/workflows/reusable-build.yml:17]: I was thinking about using this: https://github.com/marketplace/actions/repository-dispatch and calling it from arena and agents after a push in their `main`, what do you think?

**Comment by alexpalms** [comment]
> This PR substitutes all hand coded snippets in the docs with their real source code in the different repos (arena and agents).
> 
> The CICD workflow has been modified so that the reusable build can be called (as a test) from arena and agents repos every time there is a new PR (or a commit on main), in this way we can make sure the referenced source code files that are imported here are still there (otherwise the CICD will throw an error).

Now that I think about it, calling the build actions from arena and agents is not really needed as the docs are using the source code stored in the main branch, so there is no way to test the branches in PR.

**Comment by alexpalms** [comment]
> One additional thing I would like to implement is that, everytime there is a push to main in arena and or agents, a new docs build+deploy is triggered. But I am not sure if this can be done with the resuable workflow approach, i.e. adding `on: workflow call:` for the deploy CICD, wdyt @discordianfish ?

I was thinking about using this: https://github.com/marketplace/actions/repository-dispatch and calling it from arena and agents after a push in their `main`, what do you think?

**Comment by alexpalms** [comment]
> One additional thing I would like to implement is that, everytime there is a push to main in arena and or agents, a new docs build+deploy is triggered. But I am not sure if this can be done with the resuable workflow approach, i.e. adding `on: workflow call:` for the deploy CICD, wdyt @discordianfish ?

I was thinking about using this: https://github.com/marketplace/actions/repository-dispatch and calling it from arena and agents after a push in their `main`, what do you think?

**Pull request merged**